src/lib/common/include/sol-log.h

```diff
@@ -111,7 +111,7 @@ extern "C" {
 #define SOL_NULL_CHECK_GOTO(ptr, label) \
     do { \
         if (SOL_UNLIKELY(!(ptr))) { \
-            SOL_WRN("" # ptr "== NULL"); \
+            SOL_WRN("%s == NULL", # ptr); \
```

Comment:
good, but you forgot the non-goto version `SOL_NULL_CHECK()` is still subject to that.